### PR TITLE
Vid club connector

### PR DESCRIPTION
**Pull request by cgildenia**
## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [x] updates docstrings for any new functions or function arguments (if appropriate)
- [x] updates `CHANGELOG.md` with a summary of the changes

### CODE REVIEW

File: viadot/sources/velux_club.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                ['jobs', 'product', 'company', 'survey'].
+            from_date (str): Start date for the query, by default is the oldest date in the data.
+            to_date (str): End date for the query, if empty, datetime.today() will be used.
+            api_url (str): Generic part of the URL
+            region (str): Region filter for the query
+
+        Returns:
+            str: Final query with all filters added
+        """
+        if source in ["jobs", "product", "company"]:
```
[the remark this answers is not in the document]
Comment (cgildenia) [reply viadot/sources/velux_club.py:93]: The fourth source is in the elif

File: viadot/sources/velux_club.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        delta = from_date_obj - oldest_date_obj
+
+        if delta.days < 0:
+            raise historicalTooOld("from_date cannot be earlier than 2023-03-22!!")
+
+        to_date_obj = datetime.strptime(to_date, "%Y-%m-%d")
+        delta = to_date_obj - from_date_obj
+
+        if delta.days < 0:
+            raise datesNOK("to_date cannot be earlier than from_date!")
```
[the remark this answers is not in the document]
Comment (cgildenia) [reply viadot/sources/velux_club.py:145]: As all exceptions are related to validation of input, I have grouped them into ValidatioError and added the error message

File: viadot/sources/velux_club.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
+import json
+import urllib
+import os
+
+from typing import Any, Dict, List, Literal
+from datetime import datetime
+
+import pandas as pd
+
+
```
Comment (Rafalz13) [inline viadot/sources/velux_club.py:10]: Remove 1 extra line

**Review by Rafalz13 — COMMENTED** [review]
Added few comments. Please add tests.

**Review by Rafalz13 — COMMENTED** [review]
Added few comments. Please add tests.

File: viadot/tasks/vid_club.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import pandas as pd
+import prefect
+from prefect import Task
+from prefect.tasks.secrets import PrefectSecret
+from prefect.utilities import logging
+from prefect.utilities.tasks import defaults_from_attrs
+
+from ..exceptions import ValidationError
+from ..sources import VidClub
+from .azure_key_vault import AzureKeyVaultSecret
```
Comment (Rafalz13) [inline viadot/tasks/vid_club.py:16]: Please remove as you are not using `AzureKeyVaultSecret`

File: viadot/tasks/vid_club.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+from typing import List, Any, Literal, Dict
+
+import pandas as pd
+import prefect
+from prefect import Task
+from prefect.tasks.secrets import PrefectSecret
+from prefect.utilities import logging
+from prefect.utilities.tasks import defaults_from_attrs
+
+from ..exceptions import ValidationError
```
Comment (Rafalz13) [inline viadot/tasks/vid_club.py:14]: Please remove as you are not using `ValidationError`

File: viadot/tasks/vid_club.py [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+import os
+from datetime import timedelta
+from typing import List, Any, Literal, Dict
+
+import pandas as pd
+import prefect
+from prefect import Task
+from prefect.tasks.secrets import PrefectSecret
+from prefect.utilities import logging
+from prefect.utilities.tasks import defaults_from_attrs
```
Comment (Rafalz13) [inline viadot/tasks/vid_club.py:12]: Please remove as you are not using `defaults_from_attrs`

File: viadot/sources/vid_club.py [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+        Args:
+            source (str): The endpoint source to be accessed, has to be among these:
+                ['jobs', 'product', 'company', 'survey'].
+            from_date (str): Start date for the query.
+            to_date (str): End date for the query, if empty, datetime.today() will be used.
+            api_url (str): Generic part of the URL.
+            items_per_page (int): number of entries per page.
+
+        Returns:
+            str: Final query with all filters added.
```
Comment (Rafalz13) [inline viadot/sources/vid_club.py:66]: Please add `Raises` section in the docstring. In this function, you have `ValidationError`.

File: viadot/sources/vid_club.py [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+        Args:
+            source (str): The endpoint source to be accessed, has to be among these:
+                ['jobs', 'product', 'company', 'survey'].
+            from_date (str, optional): Start date for the query, by default is the oldest date in the data 2022-03-22.
+            to_date (str, optional): End date for the query. By default, datetime.today() will be used.
+            items_per_page (int, optional): Number of entries per page. 100 entries by default.
+            region (str, optinal): Region filter for the query. By default, it is empty.
+
+        Returns:
+            pd.DataFrame: Table of the data carried in the response.
```
Comment (Rafalz13) [inline viadot/sources/vid_club.py:98]: Please add `Raises` section in the docstring. In this function, you have `ValidationError`.

File: viadot/sources/vid_club.py [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+
+    """
+
+    def __init__(self, *args, credentials: Dict[str, Any] = None, **kwargs):
+        """
+        Create an instance of VidClub.
+
+        Args:
+            credentials (dict, optional): Credentials to Vid Club APIs.
+                Defaults to dictionary.
```
Comment (Rafalz13) [inline viadot/sources/vid_club.py:31]: Please add `Raises` section in the docstring. In this function, you have `CredentialError`.

File: tests/integration/test_vid_club.py [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+
+@pytest.mark.init
+def test_default_credential_param():
+    vc = VidClub()
+    assert vc.credentials != None and type(vc.credentials) == dict
+
+
+@pytest.mark.proper
+def test_build_query_wrong_source():
+    with pytest.raises(Exception):
```
Comment (Rafalz13) [inline tests/integration/test_vid_club.py:38]: Please add an expected message and correct Exception. check other tests for reference.

File: tests/integration/test_vid_club.py [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+        vc = VidClub()
+        query = vc.build_query(source="test")
+
+
+@pytest.mark.proper
+def test_get_response_wrong_source():
+    with pytest.raises(Exception):
```
Comment (Rafalz13) [inline tests/integration/test_vid_club.py:45]: Please add an expected message and correct Exception. check other tests for reference.

**Pull request merged**